src/browser/BrowserService.cpp

```diff
@@ -737,6 +738,40 @@ void BrowserService::convertAttributesToCustomData(QSharedPointer<Database> db)
     }
 }
 
+void BrowserService::performGlobalAutoType(const QString& url)
+{
+    emit osUtils->globalShortcutTriggered("autotype", url);
+}
+
+/**
+ * Gets the base domain of URL.
+ *
+ * Returns the base domain, e.g. https://another.example.co.uk -> example.co.uk
+ */
+QString BrowserService::getBaseDomainFromUrl(const QString& url) const
+{
+    QUrl qurl = QUrl::fromUserInput(url);
+    QString host = qurl.host();
+
+    // If the hostname is an IP address, return it directly
+    QHostAddress hostAddress(host);
```

Comment:
Here was a bug where we passed the URL directly to QHostAddress. This caused for example "192.168.0.1:8000" to return null value because QHostAddress doesn't handle any ports. Instead we are now passing just the hostname for checking. Test are also updated.